### PR TITLE
Feature collapsable menus

### PR DESCRIPTION
**Pull request by rwhite27**

### CODE REVIEW

File: src/Mapper/MenusMapper.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        $menusDto->id = $menus->getId();
+        $menusDto->apps_id = $menus->apps_id;
+        $menusDto->companies_id = $menus->companies_id;
+        $menusDto->name = $menus->name;
+        $menusDto->slug = $menus->slug;
+
+        foreach ($menus->getLinks() as $link) {
+
+            if($link->parent_id == 0)
```
Comment (kaioken) [inline src/Mapper/MenusMapper.php:30]: @rwhite27 isParent instead of == 0

File: src/Mapper/MenusMapper.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            {
+                $childLinks = MenusLinks::find([
+                    'conditions' => 'parent_id = ?0 and is_deleted = 0',
+                    'bind' => [$link->getId()]
+                ]);
+    
+                if(sizeof($childLinks) != 0){
```
Comment (kaioken) [inline src/Mapper/MenusMapper.php:37]: @rwhite27 $childLink would be null if you dont get a return so just validatin it should get good enought? !empty?

**Review by kaioken — CHANGES_REQUESTED** [review]

File: src/Mapper/MenusMapper.php (hunk 2 above)
Comment (rwhite27) [reply src/Mapper/MenusMapper.php:37]: does not work. I checked before resorting to this solution.

File: src/Models/MenusLinks.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        return 'menus_links';
+    }
+
+    /**
+     * Check whether or not the menuslink is a parent
+     * @return bool
+     */
+    public function isParent(): bool
+    {
+        return $this->parent_id == 0 ? true : false;
```
Comment (kaioken) [inline src/Models/MenusLinks.php:119]: @rwhite27 just return  return $this->parent_id == 0 its good enought

**Review by kaioken — CHANGES_REQUESTED** [review]

**Review by kaioken — APPROVED** [review]

**Review by kaioken — APPROVED** [review]

**Pull request merged**